src/resmom/catch_child.c

```diff
@@ -2404,9 +2404,7 @@ del_job_hw(job *pjob)
 			end_time = begin_time;
 			/* add jobid to the seed */
 			srandom((unsigned)(atoi(pjob->ji_qs.ji_jobid) + begin_time));
-			for (i=1;
-				(total_time = end_time - begin_time) < alps_release_timeout;
-				++i) {
+			for (i = 1; (total_time = end_time - begin_time) < alps_release_timeout; ++i) {
```

Comment:
Don't set variables in the conditional portion of a loop.  In this case, I don't see total_time being used, so you can just remove it.

Reply:
Well, this part of the code is not something I have changed but I'll be happy to amend it if it creates a problem. I do see total time being used to log the amount of time we waited for canceling a reservation, it is outside the loop. Since the code has been there for years and has not caused any warnings or troubles I'd argue to have it as is, but, if you feel strongly about not having to set a variable in the conditional portion, I'll move it.